### PR TITLE
Fix typing jit issue on RoIPool and RoIAlign

### LINKED ISSUE #6395: Inconsistency between RoIPool's forward and roi_pool
### 🚀 The feature

Currently the functional `roi_pool` method supports both `Tensor` and `List[Tensor]` for boxes, but the module forward function of `RoIPool` expects boxes(rois) to be `Tensor` only. Can we change the function signature to support both? 

The inconsistency can cause issue during torchscripting, for example:
```
from typing import List
import torch
from torch import Tensor
from torchvision.ops import RoIPool

class Model(torch.nn.Module):
    def __init__(self):
        super().__init__()
        self.pool = RoIPool(output_size=[3,3], spatial_scale=1.0)

    def forward(self, imgs, boxes: List[Tensor]):
        return self.pool(imgs, boxes)

model = Model()
torch.jit.script(model)
```
Raise the following error:

> forward(__torch__.torchvision.ops.roi_pool.RoIPool self, Tensor input, Tensor rois) -> Tensor:
> Expected a value of type 'Tensor' for argument 'rois' but instead found type 'List[Tensor]'.
> Empty lists default to List[Tensor]. Add a variable annotation to the assignment to create an empty list of another type (torch.jit.annotate(List[T, []]) where T is the type of elements in the list for Python 2)
> :
>   File "<ipython-input-8-9536dd2f0cc1>", line 12
>     def forward(self, imgs, boxes: List[Tensor]):
>         return self.pool(imgs, boxes)
>                ~~~~~~~~~ <--- HERE

### Motivation, pitch

Make sure the model can be scriptable when using list of boxes as the input.

### Alternatives

Only support Tensor as the input, user are required to convert bbox of different images into one tensor. Potentially make `_utils.convert_boxes_to_roi_format` public so that users don't need to write the conversion function.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by datumbox**
Fixes #6395

### CODE REVIEW

File: test/test_ops.py [hunk 1]
```diff
@@ -210,6 +219,10 @@ def get_slice(k, block):
     def test_boxes_shape(self):
         self._helper_boxes_shape(ops.roi_pool)
 
+    def test_jit_boxes_list(self):
+        model = PoolWrapper(ops.RoIPool(output_size=[3, 3], spatial_scale=1.0))
+        torch.jit.script(model)
```
Comment (vfdev-5) [inline test/test_ops.py:224]: Maybe we could also check model execution on list of tensors ?

**Review by vfdev-5 — APPROVED** [review]
LGTM. 
I added a comment to discuss

File: test/test_ops.py (hunk 1 above)
Comment (datumbox) [reply test/test_ops.py:224]: Yes good call. I'll add it now.

**Review by vfdev-5 — APPROVED** [review]
Thanks !

**Pull request merged**